File: apps/java/src/main/java/tdd/fizzbuzz/domain/type/FizzBuzzType03.java
package tdd.fizzbuzz.domain.type;

import tdd.fizzbuzz.domain.model.FizzBuzzValue;

public final class FizzBuzzType03 extends FizzBuzzType {

    @Override
    public FizzBuzzValue generate(int number) {
        if (isFizzBuzz(number)) {
            return new FizzBuzzValue(number, "FizzBuzz");
        }
        return new FizzBuzzValue(number, Integer.toString(number));
    }
}

File: apps/php/src/FizzBuzzType03.php
<?php

namespace App;

class FizzBuzzType03 implements FizzBuzzType
{
    public function generate(int $number): FizzBuzzValue
    {
        if ($number % 15 === 0) {
            return new FizzBuzzValue($number, 'FizzBuzz');
        }

        return new FizzBuzzValue($number, (string) $number);
    }
}
